### DISCUSSION TITLE
Generate simplest build.sbt with latest scalaVersion if running sbt in a dir without build.sbt

### DISCUSSION THREAD

**Discussion by bjornregnell** [sbt/sbt#6772] [Ideas]
It is possible to start `sbt` in a dir with no `build.sbt` at all which is cool for a beginner. 

But currently I get old Scala 2.12.15 and `++3.1.0!` is a really cryptic thing for a beginner and only holds for the current session. 

I think it would be awesome for a beginner if `sbt` (in addition the generated project/build.properties with latest sbt version) also create, when absent,  a `build.sbt` including only the minimal bare setting `scalaVersion := <latest>`

**Comment by bjornregnell** [comment]
See also https://discord.com/channels/632150470000902164/922600050989875282/926413393525624832

**Comment by eed3si9n** [comment]
This is what `sbt new` is for, IMO.

**Reply by bjornregnell** [reply]
Well if you don't know about the `new` arg and just installed sbt and try it out in terminal you get a really old scala repl if you start console etc. `sbt` is simpler than `sbt new` but the simplest thing currently does the "wrong" thing...

**Comment by SethTisue** [comment]
Closely related: #2123.

When no project files are present, I think it would be reasonable to bring up a prompt that asks for confirmation and also suggests `sbt new` as an alternative. This would also address #2123. (It happens to me pretty often that I accidentally run `sbt` in the wrong directory and it makes files that I then have to delete.)

**Reply by bjornregnell** [reply]
Yes that's nice. Also for a beginner this would be helpful. But if I write `sbt new` then:
```
$ sbt new
[info] welcome to sbt 1.6.1 (Eclipse Adoptium Java 11.0.13)
[info] loading settings for project global-plugins from gpg.sbt ...
[info] loading global plugins from /home/bjornr/.sbt/1.0/plugins
[info] set current project to new (in build file:/tmp/sbt_bf389b77/new/)
Template not found for: 

```
Which is reported here https://github.com/sbt/sbt/issues/6574

So I think `sbt` and `sbt new` should behave something like this:
```
$ sbt 
[info] welcome to sbt 1.6.1 (Eclipse Adoptium Java 11.0.13)
[info] loading settings for project global-plugins from gpg.sbt ...
[info] loading global plugins from /home/bjornr/.sbt/1.0/plugins
[info] set current project to new (in build file:/tmp/sbt_bf389b77/new/)
This folder has no `build.sbt` file.
Would you like to create a minimal project? Type y<Enter> for yes or <Enter> for no 
```
And then I'd just get these minimal files if pressing y+Enter:
```
$ cat build.sbt 
scalaVersion := <latest>
$ cat project/build.properties 
sbt.version=1.6.1
```
BTW: Is there a reliable way of knowing the latest scala version via some wget/curl?

**Comment by SethTisue** [comment]
> Is there a reliable way of knowing the latest scala version via some wget/curl?

you could grab https://raw.githubusercontent.com/scala/scala-lang/main/download/scala2.md and/or https://raw.githubusercontent.com/scala/scala-lang/main/download/scala3.md and look at the `release_version:`. that's probably the simplest method and the most uniform between Scala 2 and 3. but it may not be "reliable" in the sense of continuing to work months or years from now, if the website gets changed around

another way would be to ask Coursier:

* for Scala 2, `cs resolve --no-default -r central org.scala-lang:scala-library:2+` (note the `--no-default`, otherwise one of the places it will look is `~/.ivy2/local`)
* for Scala 3, nightlies are published to Maven Central (rather than to a custom resolver) so the same approach (with `org.scala-lang:scala3-library_3:3+` is liable to get you some monstrosity like `3.1.2-RC1-bin-20220101-eb75a1a-NIGHTLY`. maybe someone whose Coursier-fu is better than mine knows how to prevent that

other alternatives:

* you could use some other, non-Scala-specific tool other than Coursier that knows how to query Maven Central
* or you could make an api.github.com query (via `hub api` or more directly) to the scala/scala and/or lampepfl/dotty releases pages on GitHub

cc @dwijnand
